### PR TITLE
Nutanix field

### PR DESCRIPTION
**Pull request by sgaoshang**
Add nutanix fields check
case:
[OCP-51095 - [ipi-on-nutanix] Install cluster with invalid IPI options](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-51095)
[OCP-70747 - [ipi-on-nutanix] create multi failureDomains cluster with invalid fields](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-70747)

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-single-osimage-customized-reboot-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-single-osimage-customized-reboot-fields-f28

**Comment by sgaoshang** [comment]
@jianlinliu pls help to review it, thanks.

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml [hunk 1]
```diff
@@ -333,9 +333,9 @@ tests:
     test:
     - chain: cucushift-installer-check-cluster-health
     workflow: cucushift-installer-rehearse-ibmcloud-ipi-private-sshkey
-- as: nutanix-ipi-zones-single-osimage-customized-reboot-nodes-f28
+- as: nutanix-ipi-zones-single-osimage-customized-reboot-fields-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml:336]: I suggest to prepare a separate install workflow to cover the invalid fields testing.

File: ci-operator/step-registry/cucushift/installer/check/nutanix/fields/cucushift-installer-check-nutanix-fields-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ref:
+  as: cucushift-installer-check-nutanix-fields
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/fields/cucushift-installer-check-nutanix-fields-ref.yaml:2]:
```suggestion
  as: cucushift-installer-check-nutanix-invalid-fields
```

File: ci-operator/step-registry/cucushift/installer/check/nutanix/fields/cucushift-installer-check-nutanix-fields-commands.sh [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  grep -v "password\|username\|pullSecret\|auth" <"${dir}"/install-config.yaml
+  set +o errexit
+  openshift-install --dir="${dir}" create cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+  wait "$!"
+  ret="$?"
+  echo "Installer exit with code $ret"
+}
+
+if [ ! -f /tmp/yq ]; then
+  curl -L "https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_$(get_arch)" \
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/fields/cucushift-installer-check-nutanix-fields-commands.sh:26]: I think if use `from: upi-installer` to run the step, we can skip the yq download step.

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml [hunk 4]
```diff
@@ -347,6 +347,12 @@ tests:
     test:
     - ref: cucushift-installer-check-cluster-health
     workflow: cucushift-installer-rehearse-nutanix-ipi-zones-customized-resource
+- as: nutanix-ipi-invaid-fields-f28
+  cluster: build01
+  cron: 2 20 16 * *
+  steps:
+    cluster_profile: nutanix-qe-zone
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml:354]: Shall we use the zone cluster profile to run the testing ?

Comment (jianlinliu) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml:354]: Okay, sounds like cucushift-installer-rehearse-nutanix-ipi-invaid-fields includes some zone setting check, if that, I think we should rename the install workflow to indicated it is a zonal install workflow.

File: ci-operator/step-registry/cucushift/installer/check/nutanix/invalid-fields/cucushift-installer-check-nutanix-invalid-fields-commands.sh [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+check_result=0
+
+function run_install() {
+  echo "install-config.yaml"
+  echo "-------------------"
+  grep -v "password\|username\|pullSecret\|auth" <"${dir}"/install-config.yaml
+  set +o errexit
+  openshift-install --dir="${dir}" create cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+  wait "$!"
+  ret="$?"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/invalid-fields/cucushift-installer-check-nutanix-invalid-fields-commands.sh:16]: suggest to add the following line in the begaining of function.
```
local ret
```

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-nutanix-ipi-zones-invaid-fields-f28

**Pull request merged**